src/pages/Message.jsx

```diff
@@ -1,134 +1,191 @@
+      });
+      setMessage(msgs)
+      setMessageDocRef(documentRe)
+
+      return msgs;
+    } catch (error) {
+      console.error("Error fetching subcollection data:", error);
+      return [];
+    }
+  };
 
-    return () => {
-      isMounted = false;
-    };
-  }, [childId]);
+  const fetchNextMessage = async () => {
+    try {
+      if (!lastMessageDoc) return;
 
-const sendMessage = async (e) => {
-  e.preventDefault();
+      const subcollectionRe = collection(messageDocRef, "letters");
+      const q = query(subcollectionRe, orderBy("created_at", "desc"), startAfter(lastMessageDoc), limit(1));
+      const subcollectionSnapshott = await getDocs(q);
 
-  if (!newMessage.trim()) return;
+      if (subcollectionSnapshott.empty) {
+        console.log("No more messages available.");
+        return;
+      }
 
-  const newMessagePayload = {
-    deleted_at: null,
-    content: newMessage.trim(),
-    moderated_at: null,
-    content_type: "text",
-    sender: childRef,
+      const newMessage = subcollectionSnapshott.docs[0].data();
+      setMessage(prevMessages => [...prevMessages, newMessage]);
+      setLastMessageDoc(subcollectionSnapshott.docs[0]); // Update last document for next pagination
+    } catch (error) {
+      console.error("Error fetching next message:", error);
+    }
   };
 
-  const updatedMessages = [...message, newMessagePayload];
-
-  await updateDoc(messageDocRef.docs[0].ref, { Messages: updatedMessages });
-
-  setMessage(updatedMessages);
-
-  setNewMessage("");
-};
-return (
-  <Box>
-    <Stack direction="row" sx={{ alignItems: "center" }}>
-    </Stack>
-    <Typography variant="paragragh" paddingX={2} marginX={2}>
-      <hr />
-    </Typography>
-    <div>
-      {user ? (
-        <div>
-          <MessagesComp chat={message} />
-          <NewMessage setNewMessage={setNewMessage} sendMessage={sendMessage} newMessage={newMessage} />
-        </div>
-      ) : (
-        <div>not logged in</div>
-      )}
-    </div>
-  </Box>
-);
+  const sendMessage = async (e) => {
+    e.preventDefault();
+    if (!newMessage.trim()) return;
+    if (!userSet) await findUser()
+
+    try {
+      const newMessagePayload = {
+        deleted_at: null,
+        letter: newMessage,
+        created_at: new Date(),
+        moderation: {
+          approved: false,
+          comment: "",
+          moderated_at: null
+        },
+        sentby: user,
+        attachments: imagePreviewUrl
+      };
+      const updatedMessages = [...messages, newMessagePayload];
+      const subcollectionRef = collection(messageDocRef, "letters");
+      // REZ - this will break if permissions set
+      await addDoc(subcollectionRef, newMessagePayload);
+
+      console.log("Updated messages:", updatedMessages);
+
+      if (!messageDocRef) {
+        throw new Error("Message document reference not found.");
+      }
+
+      await getSubData()
+      setNewMessage("");
```

Comment:
Can you also have the recently send message appear in the UI?

Reply:
not sure what you mean but when we have sent a new message we getSubData so we get the messages from the db - therefore the new message will be returned too

Reply:
But when I tried sending a message to the chat, my message did not appear as the last message in the chat.

Reply:
this will be fixed when we add in the UI feedback for hitting send. So will resolve this thread